components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/OutboundProvisioningManager.java

```diff
@@ -825,7 +827,10 @@ protected boolean canUserBeProvisioned(ProvisioningEntity provisioningEntity,
         }
 
         String userName = getUserName(provisioningEntity.getAttributes());
-        List<String> roleListOfUser = getUserRoles(userName, tenantDomain);
+        List<String> roleListOfUser = Stream.concat(getUserRoles(userName, tenantDomain).stream(),
+                provisioningEntity.getAttributes().get(ClaimMapping.build
```

Comment:
what if 
provisioningEntity.getAttributes() is null ?
provisioningEntity.getAttributes().get(x) is null ?